vms/components/chain/state.go

```diff
@@ -138,11 +141,21 @@ func (s *State) initialize(config *Config) {
 
 func NewState(config *Config) *State {
 	c := &State{
-		verifiedBlocks:   make(map[ids.ID]*BlockWrapper),
-		decidedBlocks:    &cache.LRU[ids.ID, *BlockWrapper]{Size: config.DecidedCacheSize},
-		missingBlocks:    &cache.LRU[ids.ID, struct{}]{Size: config.MissingCacheSize},
-		unverifiedBlocks: &cache.LRU[ids.ID, *BlockWrapper]{Size: config.UnverifiedCacheSize},
-		bytesToIDCache:   &cache.LRU[string, ids.ID]{Size: config.BytesToIDCacheSize},
+		verifiedBlocks: make(map[ids.ID]*BlockWrapper),
+		decidedBlocks: cache.NewSizedLRU[ids.ID, *BlockWrapper](
+			config.DecidedCacheSize,
+			func(bw *BlockWrapper) int {
+				return len(bw.Bytes()) + 2*pointerOverhead
+			},
+		),
+		missingBlocks: &cache.LRU[ids.ID, struct{}]{Size: config.MissingCacheSize},
+		unverifiedBlocks: cache.NewSizedLRU[ids.ID, *BlockWrapper](
+			config.UnverifiedCacheSize,
+			func(bw *BlockWrapper) int {
+				return len(bw.Bytes()) + 2*pointerOverhead
+			},
+		),
+		bytesToIDCache: &cache.LRU[string, ids.ID]{Size: config.BytesToIDCacheSize},
```

Comment:
while we keep this cache item based, should we reduce its size? I don't expect it to make a big difference if we don't, but IIUC it'll hold infos much longer than expected now, given other caches are size based?

Reply:
Great 👁️. I had forgotten that we had block bytes as a key in a cache.